### DISCUSSION TITLE
semantic meaning of prob.value

### DISCUSSION THREAD

**Discussion by siyuanpurdue** [cvxpy/cvxpy#1290] [Q&A]
I can't find any explanation of this phenomenon while using cvxpy.
```
import cvxpy as cp
import numpy as np
def peak_estimation(Z_t):
    t = Z_t.size
    remain_trace = cp.Variable(time_slot - t)

    prob = cp.Problem(cp.Minimize(offline_opt(np.array(Z_t.tolist() + \
                      list(remain_trace)))), constraints = \
                      [lower_agg_prediction()[t:]<=remain_trace, \
                      remain_trace<= upper_agg_prediction()[t:]])
    prob.solve(solver=cp.SCS, max_iters=50000)
    
    return offline_opt(np.array(Z_t.tolist() + list(remain_trace.value)))
```
Here Z_t is np.array, offline_opt returns the solution of another optimization problem. So instead of returning by calling offline_opt, I was expecting I could call prob.value for the minimization result. But prob.value seems to be totally different from the result of calling offline_opt function. Theoretically it should be the same objective function. I am confused. P.S.: We can treat offline_opt as a black box, but the code is as follows.
```
def offline_opt(overall_load):
    ''' This calculates the offline optimal peak for a specific trace
     over_load is one 24-hour profile for the load consumption
     agg is the aggregate battery model'''
    
    # charge\dischaging decision at each time slot
    power_signal = cp.Variable(time_slot)
    
    #linear_constraint = LinearConstraint(gen_A(),gen_LB(agg),gen_UB(agg))
    
    prob = cp.Problem(cp.Minimize(max_peak(power_signal, overall_load)), \
                     constraints=[gen_LB(agg)<= gen_A()@power_signal, \
                                 gen_A()@power_signal<=gen_UB(agg)])
    prob.solve(solver=cp.SCS, max_iters=20000)
    print("power_signal.value",power_signal.value)
    return prob.value
```
So here prob.value returns the same value as max_peak(power_signal.value, overall_load)

**Comment by SteveDiamond** [comment]
Sometimes due to numerical imprecision ``prob.value`` will differ from evaluating the objective directly with ``objective.value``. Is that what you're talking about?

**Comment by siyuanpurdue** [comment]
In the first snippet, the return value (offline_opt(np.array(Z_t.tolist() + list(remain_trace.value)))) is very different from prob.value? I mean, in my example, prob.value is 30,   (offline_opt(np.array(Z_t.tolist() + list(remain_trace.value)))) returns a value of 120. So I can't imagine it's because of numerical imprecision?

**Reply by SteveDiamond** [reply]
It could actually. What problem status are you getting? Is it OPTIMAL or OPTIMAL_INACCURATE? If it's the latter there could be a substantial difference. Regardless this is a confusion we're going to sort out.

**Reply by siyuanpurdue** [reply]
If I use verbose=True, I do get: "Problem status: optimal". Does that mean it's solved perfectly despite the warning:
```
WARN: A->p (column pointers) not strictly increasing, column 20 empty
```
